src/libs/kata-types/src/config/drop_in.rs

```diff
@@ -213,3 +213,205 @@ mod toml_tree_ops {
+            dropin_dir.push("config.d");
+            let result = fs::create_dir(&dropin_dir);
+            assert!(result.is_ok());
+
+            let config = load(&config_path).unwrap();
+            assert_eq!(
+                config.hypervisor["qemu"].path,
+                "/usr/bin/qemu-kvm".to_string()
+            );
+            assert_eq!(config.hypervisor["qemu"].device_info.default_bridges, 3);
+            assert!(config.runtime.debug);
+            assert_eq!(config.runtime.internetworking_model, "tcfilter".to_string());
+        }
+
+        #[test]
+        fn test_dropins() {
+            let tmpdir = tempfile::tempdir().unwrap();
+
+            let runtime_config_file_data = r#"
+                [hypervisor.qemu]
+                path = "/usr/bin/qemu-kvm"
+                default_bridges = 3
+                [runtime]
+                enable_debug = true
+                internetworking_model="tcfilter"
+            "#;
+
+            let dropin_data = r#"
+                [hypervisor.qemu]
+                default_vcpus = 2
+                default_bridges = 4
+                shared_fs = "virtio-fs"
+                [runtime]
+                sandbox_cgroup_only=true
+                internetworking_model="macvtap"
+                vfio_mode="guest-kernel"
+            "#;
+
+            let dropin_override_data = r#"
+                [hypervisor.qemu]
+                shared_fs = "virtio-9p"
+                [runtime]
+                vfio_mode="vfio"
+            "#;
+
+            let mut config_path = PathBuf::from(tmpdir.path());
+            config_path.push("runtime.toml");
+            let result = create_file(&config_path, runtime_config_file_data.as_bytes());
+            assert!(result.is_ok());
+
+            let mut dropin_dir = PathBuf::from(tmpdir.path());
+            dropin_dir.push("config.d");
+            let result = fs::create_dir(&dropin_dir);
+            assert!(result.is_ok());
+
+            let mut dropin_path = dropin_dir.clone();
+            dropin_path.push("10-base");
+            let result = create_file(&dropin_path, dropin_data.as_bytes());
+            assert!(result.is_ok());
+
+            let mut dropin_override_path = dropin_dir.clone();
+            dropin_override_path.push("20-override");
+            let result = create_file(&dropin_override_path, dropin_override_data.as_bytes());
+            assert!(result.is_ok());
+
+            let config = load(&config_path).unwrap();
+            assert_eq!(
+                config.hypervisor["qemu"].path,
+                "/usr/bin/qemu-kvm".to_string()
+            );
+            assert_eq!(config.hypervisor["qemu"].cpu_info.default_vcpus, 2);
+            assert_eq!(config.hypervisor["qemu"].device_info.default_bridges, 4);
+            assert_eq!(
+                config.hypervisor["qemu"].shared_fs.shared_fs,
+                Some("virtio-9p".to_string())
+            );
```

Comment:
This does an allocation of a new copy of the "virtio-9p" string, which is a little bit ugly, though it doesn't really matter for a test case.  If you wanted to, the way to do this comparison without the allocation is:
```
assert_eq!(config.hypervisor["qemu"].shared_fs.shared_fs.as_ref(), Some("virtio-9p"));
```